casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -293,6 +305,19 @@ object ProtoUtil {
       StringValue.of(blockMessage.shardId).toByteArray,
       blockMessage.extraBytes.toByteArray
     )
+
+  /**
+    * Create hash of a BlockMessage, all fields must be included except signature
+    */
+  def hashBlockVersion2(blockMessage: BlockMessage): BlockHash = {
+    val emptyBytes = ByteString.EMPTY
+    val blockClearSigData = blockMessage.copy(
+      blockHash = emptyBytes,
+      sig = emptyBytes
```

Comment:
Sig algorithm is still inside signed body? Did we want to make it out?

Reply:
We can do it as part of 0.13 version. I was trying to keep minimal changes here.